Quadriga/src/main/java/edu/asu/spring/quadriga/web/workspace/ConceptCollectionWorkspaceController.java

```diff
@@ -32,270 +30,202 @@
-			IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId,userId);
-			model.addAttribute("workspacedetails", workspace);
-			model.addAttribute("projectid", workspaceId);
-			return "auth/workbench/workspace/conceptcollections";
-		} else {
-			for (int i = 0; i < values.length; i++) {
-				try {
-			       workspaceCCManager.deleteWorkspaceCC(workspaceId, userId, values[i]);
-				} catch (QuadrigaStorageException e) {
-					logger.error("Issue while accessing DB",e);
-				}
-				if(!msg.equals("")){
-					flag=1;
-				}
-			}
-		}
-		if(flag==0){
-			model.addAttribute("deletesuccess", 1);
-		}else{
-			model.addAttribute("deletesuccess", 0);
-		}
-		List<IWorkspaceConceptCollection> conceptCollectionList = null;
-		try {
-			conceptCollectionList = workspaceCCManager.listWorkspaceCC(workspaceId, userId);
-		} catch (QuadrigaStorageException e) {
-			logger.error("Issue while accessing DB",e);
-		}
-		if(conceptCollectionList == null){
-			logger.info("Dictionary list is empty buddy");
-		}
-		model.addAttribute("conceptCollectionList", conceptCollectionList);
-		IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId,userId);
-		model.addAttribute("workspacedetails", workspace);
-		model.addAttribute("projectid", workspaceId);
-		return "auth/workbench/workspace/conceptcollections";
-	}
+        String[] values = req.getParameterValues("selected");
+        if (values == null) {
+            attr.addFlashAttribute("show_error_alert", true);
+            attr.addFlashAttribute("error_alert_msg", "Please select a Concept Collection");
+            return "redirect:/auth/workbench/workspace/" + workspaceId + "/addconceptcollection";
+        }
+        for (int i = 0; i < values.length; i++) {
+            msg = workspaceCCManager.addWorkspaceCC(workspaceId, values[i], userId);
+            if (!msg.equals("")) {
+                flag = 1;
+            }
+        }
+        if (flag == 0) {
+            attr.addFlashAttribute("show_success_alert", true);
+            attr.addFlashAttribute("success_alert_msg", "Concept Collection added to workspace successfully");
+        } else {
+            attr.addFlashAttribute("show_error_alert", true);
+            attr.addFlashAttribute("error_alert_msg", "Concept Collection is already added or some internal issue");
+        }
+        List<IWorkspaceConceptCollection> conceptCollectionList = null;
+        conceptCollectionList = workspaceCCManager.listWorkspaceCC(workspaceId, userId);
+        if (conceptCollectionList == null) {
+            logger.info("conceptCollectionList list is empty buddy");
+        }
+        model.addAttribute("conceptCollectionList", conceptCollectionList);
+        IWorkSpace workspace = wsManager.getWorkspaceDetails(workspaceId, userId);
+        model.addAttribute("workspacedetails", workspace);
+        model.addAttribute("workspaceId", workspaceId);
+        return "redirect:/auth/workbench/workspace/" + workspaceId + "/addconceptcollection";

```

Comment:
if you redirect you don't need to add the attributes to the model, they won't be available in the page anyways (that's what you use the redirectAttributes for).